src/main/java/com/aws/greengrass/security/SecurityService.java

```diff
@@ -163,14 +166,53 @@ public KeyManager[] getKeyManagers(URI privateKeyUri, URI certificateUri)
         return provider.getKeyManagers(privateKeyUri, certificateUri);
     }
 
+    /**
+     * Get certificate using private and certificate URIs.
+     *
+     * @param privateKeyUri  private key URI
+     * @param certificateUri certificate URI
+     * @return X509Certificate that corresponds to the given private key and certificate URIs
+     * @throws ServiceUnavailableException if crypto key provider service is unavailable
+     * @throws KeyLoadingException         if crypto key provider service fails to load key
+     */
+    public X509Certificate getCertificate(URI privateKeyUri, URI certificateUri)
+            throws ServiceUnavailableException, KeyLoadingException {
+        logger.atTrace().kv(KEY_URI, privateKeyUri).kv(CERT_URI, certificateUri)
+                .log("Get certificate by private key and certificate URIs");
+        CryptoKeySpi provider = selectCryptoKeyProvider(privateKeyUri);
+        try {
+            return provider.getCertificate(privateKeyUri, certificateUri);
+        } catch (UnsupportedOperationException e) {
+            KeyManager[] km = provider.getKeyManagers(privateKeyUri, certificateUri);
+            if (km.length != 1 || !(km[0] instanceof X509KeyManager)) {
+                return null;
+            }
+
+            X509KeyManager x509KeyManager = (X509KeyManager) km[0];
+            KeyPair keyPair = provider.getKeyPair(privateKeyUri, certificateUri);
+            String[] aliases = x509KeyManager.getClientAliases(keyPair.getPublic().getAlgorithm(), null);
+            for (String alias : aliases) {
+                if (x509KeyManager.getPrivateKey(alias) != keyPair.getPrivate()) {
+                    continue;
+                }
+                X509Certificate[] certificateChain = x509KeyManager.getCertificateChain(alias);
+                if (certificateChain.length >= 1) {
```

Comment:
if the chain is greater, how do we know which of the multiple to return? Can you check that the certificate's public key matches the public/private key we want?

Reply:
Noticed a discussion around cert vs cert chain.

Don't assume there's only 1 cert. Why would that be the case? W.r.t. which cert, there's convention. No need for fancy logic here. Keep it simple

Reply:
I'll change this API to get the cert chain. Any component that just needs the certificate can retrieve the first one in the array as long as we can impose the order `core CA cert -> intermediate cert (if any) -> root cert` while storing the cert chain in the key store. Any other opinions?